dask/dataframe/io/tests/test_parquet.py

```diff
@@ -938,6 +938,60 @@ def test_to_parquet_default_writes_nulls(tmpdir):
     assert table[1].null_count == 2
 
 
+def test_to_parquet_consistent_schema_partition_pyarrow(tmpdir, engine):
+    check_fastparquet()
+    check_pyarrow()
+    fn = str(tmpdir.join("dask.pa"))
+
+    arrays = np.array([0, 1, 2, 4])
+    strings = np.array(["a", "b", "c", "d"])
+
+    df = pd.DataFrame([0, 0, 1, 1], columns=["partition_column"])
+    df["arrays"] = pd.Series(arrays)
+    df["strings"] = pd.Series(strings)
+
+    ddf = dd.from_pandas(df, npartitions=2)
+    ddf.to_parquet(fn, engine=engine, partition_on=["partition_column"])
+    ddf2 = dd.read_parquet(fn, engine=engine)
+
+    assert np.array_equal(ddf2["arrays"], arrays)
+    assert np.array_equal(ddf2["strings"], strings)
+    assert np.array_equal(ddf2["partition_column"], df["partition_column"])
+
+
+def test_to_parquet_inconsistent_schema_partition_pyarrow(tmpdir):
+    check_fastparquet()
+    check_pyarrow()
+    fn = str(tmpdir.join("dask.pa"))
+
+    arrays = np.array([np.array([0, 1, 2]), np.array([3, 4]), np.nan, np.nan])
+    strings = np.array(["a", "b", np.nan, np.nan])
+
+    df = pd.DataFrame([0, 0, 1, 1], columns=["partition_column"])
+    df["arrays"] = pd.Series(arrays)
+    df["strings"] = pd.Series(strings)
+
+    ddf = dd.from_pandas(df, npartitions=2)
+    schema = pa.schema(
+        [
+            ("arrays", pa.list_(pa.int64())),
+            ("strings", pa.string()),
+            ("partition_column", pa.int64()),
+        ]
+    )
+    engine = "pyarrow"
+    ddf.to_parquet(fn, engine=engine, partition_on=["partition_column"], schema=schema)
+    ddf2 = dd.read_parquet(fn, engine=engine)
+
+    # np.isnan not support for dtype "object"
+    isnan = [str(a) == "nan" for a in arrays]
+    expected_arrays = np.where(isnan, None, arrays)
+    resulting_arrays = ddf2.compute()["arrays"].values
+    for i in range(resulting_arrays.size):
+        assert np.array_equal(resulting_arrays[i], expected_arrays[i])
+    assert np.array_equal(ddf2["partition_column"], df["partition_column"])
```

Comment:
Why the conversion to numpy arrays here?  I'm curious what these lines are trying to test.

Reply:
ping @mmccarty